lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -1181,8 +1182,12 @@ impl Instance {
             Ok(())
         })?;
 
+        self.kill_state.schedule(unsafe { pthread_self() });
+
         let res = f(self);
 
+        self.kill_state.deschedule();
```

Comment:
I like this change a lot! Happy to see the `KillState` (de)scheduled here, rather than out in `swap_and_return`.